CleverFerret/src/main/java/com/universalmedialibrary/ui/music/ArtistDetailScreen.kt

```
@@ -442,16 +441,36 @@ class ArtistDetailViewModel @Inject constructor(
         }
     }
 
     fun shuffleArtist(artist: Artist) {
-        val uris = artist.tracks.shuffled().map { it.uri }
-        if (uris.isNotEmpty()) {
-            playback.setQueue(uris, 0, true)
+        viewModelScope.launch {
+            val tracks = artist.tracks.shuffled()
+            if (tracks.isNotEmpty()) {
+                val firstTrack = tracks.first()
+                musicPlayerService.playTrackFromUri(
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)